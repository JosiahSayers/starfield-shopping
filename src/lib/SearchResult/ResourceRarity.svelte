<script lang="ts">
  export let rarity: string;
  $: colorClass = ["rare", "unique", "uncommon", "exotic"].includes(
    rarity.toLowerCase()
  )
    ? rarity.toLowerCase()
    : "badge-neutral";
</script>

<span class={`badge ${colorClass} m-1`}>{rarity}</span>

<style>
  .rare {
    background-color: #a179d1;
    border-color: #a179d1;
    color: white;
  }

  .unique {
    background-color: #6fb374;
    border-color: #6fb374;
    color: white;
  }

  .uncommon {
    background-color: #6db5c5;
    border-color: #6db5c5;
    color: white;
  }

  .exotic {
    background-color: #cbb784;
    border-color: #cbb784;
    color: white;
  }
</style>
