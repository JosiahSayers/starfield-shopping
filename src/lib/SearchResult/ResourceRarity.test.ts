import { describe, afterEach, it, expect } from 'vitest';
import { cleanup, render, screen } from '@testing-library/svelte'
import ResourceRarity from './ResourceRarity.svelte'

describe('ResourceRarity', () => {
  afterEach(() => cleanup())

  it('mounts', () => {
    const { container } = render(ResourceRarity, { rarity: '' });
    expect(container).toBeTruthy();
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders the rarity', () => {
    const { container } = render(ResourceRarity, { rarity: 'test-rarity' });
    expect(container.querySelector('.badge')?.textContent).toBe('test-rarity');
  });

  describe('classes for each rarity', () => {
    it('sets the correct class for "rare"', () => {
      const { container } = render(ResourceRarity, { rarity: 'rare' });
      const classList = container.querySelector('.badge')?.classList ?? [];
      expect(Array.from(classList)).toContain('rare');
    });

    it('sets the correct class for "unique"', () => {
      const { container } = render(ResourceRarity, { rarity: 'unique' });
      const classList = container.querySelector('.badge')?.classList ?? [];
      expect(Array.from(classList)).toContain('unique');
    });

    it('sets the correct class for "uncommon"', () => {
      const { container } = render(ResourceRarity, { rarity: 'uncommon' });
      const classList = container.querySelector('.badge')?.classList ?? [];
      expect(Array.from(classList)).toContain('uncommon');
    });

    it('sets the correct class for "exotic"', () => {
      const { container } = render(ResourceRarity, { rarity: 'exotic' });
      const classList = container.querySelector('.badge')?.classList ?? [];
      expect(Array.from(classList)).toContain('exotic');
    });

    it('sets a default class', () => {
      const { container } = render(ResourceRarity, { rarity: 'common' });
      const classList = container.querySelector('.badge')?.classList ?? [];
      expect(Array.from(classList)).not.toContain('common');
      expect(Array.from(classList)).toContain('badge-neutral');
    });
  });
});
