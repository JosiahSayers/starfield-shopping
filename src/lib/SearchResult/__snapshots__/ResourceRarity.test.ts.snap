// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResourceRarity > mounts 1`] = `"<div><span class=\\"badge badge-neutral m-1 svelte-1x1khpc\\"></span></div>"`;
